codice/ddf
# DDF-2034 Updated Parameter Configuration for Content Directory Monitor

#### What does this PR do?
This PR augments the existing Content Directory parameter passing code in the following ways:
1) Prevents the parameters being added to the CreateStorageRequest so that they are not exposed to ingest plugins
2) Parses parameters according to the attribute type, and ignores attributes that fail to parse
2) Overrides valid attributes on metacards with the passed parameters
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@jrnorth
@dcruver
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire
#### How should this be tested?
Build, Install, Configure the Content Directory Monitor with an overridable attribute, ingest some data, and verify the metacard attribute is overridden after ingest.
#### Any background context you want to provide?
After discussion with @michaelmenousek, @kcwire and @jlcsmith, it was determined that this approach for metacard attribute overrides was the most logical.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2034
#### Screenshots (if appropriate)
#### Checklist:
